src/app/screens/sendStx/steps/Step1SelectRecipient.tsx

```diff
@@ -0,0 +1,146 @@
+import InputScreen from '@components/inputScreen';
+import { useBnsName, useBnsResolver } from '@hooks/queries/useBnsName';
+import useDebounce from '@hooks/useDebounce';
+// import { useBnsResolver } from '@hooks/queries/useBnsName';
+// import useDebounce from '@hooks/useDebounce';
+import useWalletSelector from '@hooks/useWalletSelector';
+import { validateStacksAddress } from '@stacks/transactions';
+import Button from '@ui-library/button';
+import Input from '@ui-library/input';
+import { FeedbackVariant } from '@ui-library/inputFeedback';
+import { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import styled from 'styled-components';
+
+type StxRecipientScreenProps = {
+  recipientAddress: string;
+  setRecipientAddress: (address: string) => void;
+  // recipientDomain: string;
+  setRecipientDomain: (address: string) => void;
+  memo: string;
+  setMemo: (memo: string) => void;
+  onNext: () => void;
+  isLoading: boolean;
+  header?: React.ReactNode;
+};
+
+const MemoInput = styled(Input)`
+  input {
+    height: 64px;
+    padding-top: 0px;
+  }
+`;
+
+const RecipientInput = styled(Input)`
+  min-height: 125px;
+`;
+
+interface InputFeedback {
+  variant: FeedbackVariant;
+  message: string;
+}
+
+function Step1SelectRecipient({
+  recipientAddress,
+  setRecipientAddress,
+  // recipientDomain,
+  setRecipientDomain,
+  memo,
+  setMemo,
+  onNext,
+  isLoading,
+  header,
+}: StxRecipientScreenProps) {
+  const { t } = useTranslation('translation', { keyPrefix: 'SEND' });
+  const { stxAddress } = useWalletSelector();
+  const [inputFeedback, setInputFeedback] = useState<InputFeedback[] | undefined>();
+  const [recipient, setRecipient] = useState(recipientAddress);
+
+  const debouncedSearchTerm = useDebounce(recipient, 300);
+  const associatedAddress = useBnsResolver(debouncedSearchTerm, stxAddress, 'STX');
+  const associatedDomain = useBnsName(debouncedSearchTerm);
+
+  useEffect(() => {
+    setRecipientAddress(recipientAddress);
+    setInputFeedback(undefined);
+
+    if (associatedAddress !== '') {
+      setRecipientAddress(associatedAddress);
+      setRecipientDomain(recipient);
+      setInputFeedback([
+        { variant: 'checkmark', message: t('ASSOCIATED_ADDRESS') },
+        { variant: 'plainIndented', message: associatedAddress },
+      ]);
+    }
```

Comment:
nit: avoid calling set state twice

```suggestion
    if (associatedAddress !== '') {
      setRecipientAddress(associatedAddress);
      setRecipientDomain(recipient);
      setInputFeedback([
        { variant: 'checkmark', message: t('ASSOCIATED_ADDRESS') },
        { variant: 'plainIndented', message: associatedAddress },
      ]);
    } else {
      setRecipientAddress(recipientAddress);
      setInputFeedback(undefined);
    }
```

Reply:
I also don't really think this is needed: 
`setRecipientAddress(recipientAddress);`

Reply:
This useEffect and the other one both call  setRecipientDomain, setRecipientAddress and setInputFeedback. Since they are dependent on different other hooks (useBnsResolver and useBnsName), isn't there a race condition since both those hooks make api calls?

Maybe the 2 useEffects need to be combined into 1 which depends on associatedAddress and associatedDomain?